cardano-cli/test/cardano-cli-golden/Test/Golden/CreateStaked.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Test.Golden.CreateStaked where
+
+import           Cardano.Api.Shelley (ShelleyGenesis (sgNetworkMagic, sgStaking))
+
+import           Cardano.Ledger.Crypto (StandardCrypto)
+import           Cardano.Ledger.Shelley.Genesis (ShelleyGenesisStaking (sgsPools, sgsStake))
+
+import           Control.Monad (filterM, void)
+import           Control.Monad.IO.Class
+import qualified Data.Aeson as Aeson
+import qualified Data.ByteString.Lazy as LBS
+import           Data.List (intercalate, sort)
+import           System.Directory
+import           System.FilePath
+
+import           Test.Cardano.CLI.Util (execCardanoCLI)
+
+import           Hedgehog (Property)
+import qualified Hedgehog as H
+import           Hedgehog.Extras (moduleWorkspace, propertyOnce)
+import qualified Hedgehog.Extras as H
+import qualified Hedgehog.Extras.Test.Golden as H
+
+
+{- HLINT ignore "Use camelCase" -}
+
+-- | Given a root directory, returns files within this root (recursively)
+tree :: FilePath -> IO [FilePath]
+tree root = do
+  -- listDirectory returns a path relative to 'root'. We need to prepend
+  -- root to it for queries below.
+  content <- map (root </>) <$> listDirectory root
```

Comment:
@carbolymer> actually it's the caller that postprocesses the result of `tree` to make it stable, along with normalization and making it relative (to avoid depending on a temporary directory name) :+1: